redbox-core/redbox/chains/components.py

```diff
@@ -1,20 +1,60 @@
+import logging
+import os
+
+from langchain_community.chat_models import ChatLiteLLM
 from langchain_elasticsearch import ElasticsearchRetriever
 from langchain_core.embeddings import Embeddings, FakeEmbeddings
-from langchain_openai import AzureChatOpenAI
 from langchain_openai.embeddings import AzureOpenAIEmbeddings, OpenAIEmbeddings
 from langchain_core.utils import convert_to_secret_str
 import tiktoken
 
+from redbox.api.callbacks import LoggerCallbackHandler
+from redbox.models.chain import AISettings
 from redbox.models.settings import Settings
 from redbox.retriever import AllElasticsearchRetriever, ParameterisedElasticsearchRetriever
-
-
-def get_chat_llm(env: Settings):
-    return AzureChatOpenAI(
-        api_key=convert_to_secret_str(env.azure_openai_api_key),
-        azure_endpoint=env.azure_openai_endpoint,
-        model=env.azure_openai_model,
-    )
+from langchain_core.language_models.fake_chat_models import GenericFakeChatModel
+
+log = logging.getLogger()
+
+
+def get_llm(ai_settings: AISettings) -> ChatLiteLLM:
```

Comment:
We shouldn't make the LLM object every time we need one. Adding functools.lru_cache(4) should means we only get a different one if the settings change up to some max (4 here).

Might have to set AISettings to be frozen to be hashable for lru_cache to work but that's what we want anyway